### PR TITLE
script improvements and docker files

### PR DESCRIPTION
**Pull request by abellmann**
Hi,

I did some fixes for some errors I discovered over time. 

I also added docker functionality to simplify the usage of this script.

Hope this helps..

Best regards
Andreas

### CODE REVIEW

**Comment by kautsig** [comment]
Hi @abellmann ,

first, sorry for the late reply. i really appreciate your work, those improvements are highly welcome!
My time is very limited at the moment, but I'll try to check the PR in detail during the weekend.

Regards,
Chris

**Comment by abellmann** [comment]
Hi,

Please let me know, if I can help making the pull request easier.. Perhaps splitting it up might help?

Regards,
Andreas

File: ews-fetch-calendar.py [hunk 1]
```diff
@@ -80,9 +98,47 @@ def print_orgmode_entry(subject, start, end, location, response):
...
+myopts, args = getopt.getopt(sys.argv[1:],"o:")
+
+
+###############################
+# o == option
+# a == argument passed to the o
+###############################
+for o, a in myopts:
+    if o == '-o':
+        ofile=a
```
Comment (kautsig) [inline ews-fetch-calendar.py:132]: This seems like dead code, because the `ofile` is never used.

File: ews-fetch-calendar.py [hunk 2]
```diff
@@ -80,9 +98,47 @@ def print_orgmode_entry(subject, start, end, location, response):
 
   print ""
 
-# Debug code
-# print_orgmode_entry("subject", "2012-07-27T11:10:53Z", "2012-07-27T11:15:53Z", "location", "participants")
-# exit(0)
+@contextmanager
+def stdout_redirected(new_stdout):
```
Comment (kautsig) [inline ews-fetch-calendar.py:102]: This seems like dead code, `stdout_redirected` is never used.

**Comment by kautsig** [comment]
Hi @abellmann!

Thanks for getting back to me. I want to apologize for being so extremely unresponsive.

I checked the docker functionality, works nicely.

I also added some remarks about the python code... I guess the stdout redirection in the python script is incomplete or was commited by accident, because `run-cron.sh` also does not use the `-o` flag. 

Regards,
Chris

File: cron-setup/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+version: '2'
+services:
+  ews-orgmode:
+    build:
+      context: .
+    image: ews-orgmode-cron
+    volumes:
+      - ../../.ews-orgmode:/data
```
Comment (kautsig) [inline cron-setup/docker-compose.yml:8]: I guess it would be good to mention that in the README, this encodes your local directory structure.

**Comment by abellmann** [comment]
It seems, as if some changes were lost due to a merge of your code changes into my fork. I have to review and fix this.

File: cron-setup/docker-compose.yml (hunk 3 above)
Comment (abellmann) [reply cron-setup/docker-compose.yml:8]: This is already documented in the README. Perhaps it is a bit unclear?

Comment (kautsig) [reply cron-setup/docker-compose.yml:8]: Apologies, I overlooked that.

**Pull request closed without merging**